Automattic/jetpack
# Pre-Push Hook: Skip Changelog Check for Release Branch

Fixes https://github.com/Automattic/jetpack/issues/20616

#### Changes proposed in this Pull Request:

* When we're using the release script, we don't want the pre-push hook to check for changelog files if we're on the release branch. 

Did my best to figure out the closest match to a potential release branch, but if there's a better way to check for this besides Regex, let me know! 

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No. 

#### Testing instructions:






Kinda tricky to test, but as long as CI checks pass, merge this PR and then run a release script when standing on `master`, like `tools/create-release-branch.sh boost 1.3.0-beta` 
* The pre-push hook should skip the changelog. 
* If you want to make sure the release branch doesn't actually get pushed, you can add `process.exitCode = 1;` to the end of `tools/js-tools/gh-hooks/pre-push-hook.js` which will abort the push after all checks complete.
